src/main/java/com/leff/midi/MidiTrack.java

```diff
@@ -112,207 +99,51 @@ private void readTrackData(byte[] data) throws IOException {
             }
             mEvents.add(E);
         }
+
+        mSize = 0;

```

Comment:
migrated from `recalculateSize` so that we calculate the size once, on read.  The object is thereafter immutable.